gm4_boat_chests/data/gm4_boat_chests/functions/main.mcfunction

```diff
@@ -0,0 +1,4 @@
+
+schedule function gm4_boat_chests:main 2s
+
+execute as @e[type=boat,tag=!gm4_bc_boat] at @s if entity @e[type=chest_minecart,distance=..2,tag=!gm4_bc_chest] unless data entity @s Passengers[0] run function gm4_boat_chests:merge
```

Comment:
```suggestion
schedule function gm4_boat_chests:main 16t
execute as @e[type=boat,tag=!gm4_bc_boat] at @s if entity @e[type=chest_minecart,distance=..2,predicate=!gm4_boat_chests:is_passenger,limit=1] unless data entity @s Passengers[0] run function gm4_boat_chests:merge
```
Gamemode 4 modules run every 16 ticks, not every 2 seconds.

Additionally I've made a slight change: technically other data packs or command users could summon a chest minecart riding another entity (for example, a chest minecart riding a creeper, cuz why not). Your code will currently dismount the chest minecart in favor of riding the boat, when normally it should be the creeper that gets in the boat and the chest minecart should stay on top of the creeper. Also, since you exclude anything with the tag `gm4_bc_chest`, a chest minecart that was previously placed in a boat that has since been broken cannot be placed in a new boat, so I remove that check.

I fix both of these issues in one swoop by including a `predicate=!gm4_boat_chests:is_passenger` in the chest minecart selector, which will exclude minecarts currently riding another entity but include minecarts that used to be in a boat chest but have since been taken out. The contents of the predicate should be as follows:
```json
{
    "condition": "minecraft:entity_properties",
    "entity": "this",
    "predicate": {
        "vehicle": {}
    }
}
```